pages/api/signup.ts

```diff
@@ -2,51 +2,54 @@ import { NextApiRequest, NextApiResponse } from "next";
-        let diet: string = req.body.diet();;
-        let other: string = req.body.other();;
+        let firstName: string = req.body.firstName;
+        let lastName: string = req.body.lastName;
+        let email: string = req.body.email;
+        let laptop: boolean = req.body.laptop;
+        let experience: string = req.body.experience;
+        let grade: number = req.body.grade;
+        let shirt: string = req.body.shirt;
+        let diet: string = req.body.diet;
+        let other: string = req.body.other;
+        let errors = []; //grouping: names, email, long paragraphs, button choices
         const names: string[] = [firstName, lastName];
         const explains: string[] = [experience, diet, other];
-        const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2-4}$/;
-        const namePattern = /^[a-zA-Z]{2-25}$/;
+        const choices: any[] = [laptop, grade, shirt];
+        const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        const namePattern = /^[a-zA-Z]{2,25}$/;
         const explainPattern = /^[a-zA-Z0-9\(\)._ -]+$/;
 
-        if (grade == null || laptop == null || shirt == null) {
-            res.status(400).json({ message: "Bad Request" })
-        } else {
-            res.status(200).json({ message: "OK" })
+        for (var s of names) {
+            if (namePattern.test(s)) {
+                errors.push(res.status(200).json({ message: "OK" }));
+            } else {
+                errors.push(res.status(400).json({ message: "Bad Request" }));
+            }
         }
 
         if (emailPattern.test(email)) {
-            res.status(200).json({ message: "OK" })
+            errors.push(res.status(200).json({ message: "OK" }));
         } else {
-            res.status(400).json({ message: "Bad Request" })
+            errors.push(res.status(400).json({ message: "Bad Request" }));
         }
 
-        for (var s in names) {
-            if (namePattern.test(s)) {
-                res.status(200).json({ message: "OK" })
+        for (var s of explains) {
+            if (explainPattern.test(s)) {
+                errors.push(res.status(200).json({ message: "OK" }));
             } else {
-                res.status(400).json({ message: "Bad Request" })
+                errors.push(res.status(400).json({ message: "Bad Request" }));
             }
         }
 
-        for (var s in explains) {
-            if (s.length > 250) {
-                res.status(400).json({ message: "Bad Request" })
-            }
-            if (explainPattern.test(s)) {
-                res.status(200).json({ message: "OK" })
+        for (let i = 0; i < 3; i++) {
+            if (choices[i] != null) {
+                errors.push(res.status(200).json({ message: "OK" }));
             } else {
-                res.status(400).json({ message: "Bad Request" })
+                errors.push(res.status(400).json({ message: "Bad Request" }));
             }
         }
+
+        return errors;
```

Comment:
Instead of the "return errors;", you can return "errors" using the res.status(200).json.

Also, for this, you can have an if statement so that if the errors array is empty, you send back a success message w/ the data.